cd2024bfs2g1-model/src/main/java/com/campusdual/cd2024bfs2g1/model/core/service/RouteService.java

```diff
@@ -0,0 +1,102 @@
+package com.campusdual.cd2024bfs2g1.model.core.service;
+
+import com.campusdual.cd2024bfs2g1.api.core.service.IRouteService;
+import com.campusdual.cd2024bfs2g1.model.core.dao.ImageDao;
+import com.campusdual.cd2024bfs2g1.model.core.dao.ImageRouteDao;
+import com.campusdual.cd2024bfs2g1.model.core.dao.LandmarkDao;
+import com.campusdual.cd2024bfs2g1.model.core.dao.RouteDao;
+import com.ontimize.jee.common.dto.EntityResult;
+import com.ontimize.jee.common.dto.EntityResultMapImpl;
+import com.ontimize.jee.common.services.user.UserInformation;
+import com.ontimize.jee.server.dao.DefaultOntimizeDaoHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Lazy
+@Service("RouteService")
+public class RouteService  implements IRouteService {
+
+
+    private RouteDao routeDao;
+    private LandmarkDao landmarkDao;
+    private ImageDao imageDao;
+    private ImageRouteDao image_routeDao;
+    private DefaultOntimizeDaoHelper daoHelper;
+
+    @Autowired
+    public RouteService(RouteDao routeDao, LandmarkDao landmarkDao, ImageDao imageDao, ImageRouteDao image_routeDao, DefaultOntimizeDaoHelper daoHelper) {
+        this.routeDao = routeDao;
+        this.landmarkDao = landmarkDao;
+        this.imageDao = imageDao;
+        this.image_routeDao = image_routeDao;
+        this.daoHelper = daoHelper;
+    }
+
+    @Override
+    public EntityResult routeQuery(Map<String, Object> keyMap, List<String> attrList) {
+        return this.daoHelper.query(routeDao, keyMap, attrList);
+
+    }
+
+
+
+    @Override
+    public EntityResult routeInsert(Map<String, Object> attrMap) {
+        UserInformation userInformation =(UserInformation) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+        if(userInformation.getAuthorities().stream().anyMatch(c->c.toString().equals("manager"))){
+            Map<String,Object> imageMapAttr=new HashMap();
+
+            EntityResult route_id_entity = this.daoHelper.insert(routeDao, attrMap);
+            if (route_id_entity.getCode()!=EntityResult.OPERATION_WRONG) {
+                imageMapAttr.put("img_code",attrMap.get("images"));
```

Comment:
utilizade as constantes declaradas no DAO correspondente para facer referencia os nomes das columnas
```suggestion
                imageMapAttr.put(ImageDao.ATTR_IMAGE_ID,attrMap.get("images"));
```